src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -341,21 +338,31 @@ object Group extends StrictLogging {
     }
   }
 
-  case class noEnforceRoleUpdate(originalRootGroup: RootGroup, updatedGroupId: Option[AbsolutePathId]) extends Validator[Option[Boolean]] {
-
-    override def apply(maybeNewEnforceRole: Option[Boolean]): Result = {
-      val originalGroup = updatedGroupId.flatMap(originalRootGroup.group) // TODO: why is groupUpdate.id optional? What is the semantic there?
-      (maybeNewEnforceRole, originalGroup.map(_.enforceRole)) match {
-        case (None, None) => Success
-        case (Some(newEnforceRole), None) if originalGroup.nonEmpty =>
-          Failure(Set(RuleViolation(maybeNewEnforceRole, s"enforce role cannot be updated to $newEnforceRole for $updatedGroupId. Use a partial update instead.")))
-        case (None, Some(_)) =>
-          Failure(Set(RuleViolation(maybeNewEnforceRole, s"enforce role cannot be removed from $updatedGroupId.")))
-        case (Some(newEnforceRole), Some(oldEnforceRole)) =>
-          if (newEnforceRole == oldEnforceRole) Success
-          else Failure(Set(RuleViolation(maybeNewEnforceRole, s"enforce role cannot be updated from $oldEnforceRole to $newEnforceRole for $updatedGroupId. Use a pratial update instead.")))
-        case _ => Success
+  case class disallowEnforceRoleChangeIfServicesChanged(originalRootGroup: RootGroup, base: AbsolutePathId, servicesGloballyModified: Boolean) extends Validator[raml.GroupUpdate] {
+    import disallowEnforceRoleChangeIfServicesChanged.EnforceRoleCantBeChangedMessage
+
+    override def apply(group: raml.GroupUpdate): Result = {
+      if (servicesGloballyModified == false) {
```

Comment:
Is there a specific reason to compare with == false instead of negating the boolean?

```suggestion
      if (!servicesGloballyModified) {
```